include/deepworks/nn.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <deepworks/placeholder.hpp>
+#include <deepworks/layer_info.hpp>
+#include <deepworks/call.hpp>
+
+namespace deepworks {
+
+template <typename D>
+struct BaseOp {
+    BaseOp(LayerInfo info) : m_info(std::move(info)) { }
+    Placeholder operator()(Placeholder in) {
+        Call call{m_info};
+        call.pass({in});
+        return call.create(static_cast<D*>(this)->output_shape(in.shape()));
```

Comment:
Should we add base method for output_shape?

Reply:
No, method always taken from `Derived`. The basic approach for `CRTP` pattern